docs/source/java/cdata.rst

```diff
@@ -0,0 +1,212 @@
+    #include <arrow/c/bridge.h>
+
+    using arrow::Int64Builder;
+
+    void FillInt64Array(const uintptr_t c_schema_ptr, const uintptr_t c_array_ptr) {
+        arrow::Int64Builder builder;
+        builder.Append(1);
+        builder.Append(2);
+        builder.Append(3);
+        builder.AppendNull();
+        builder.Append(5);
+        builder.Append(6);
+        builder.Append(7);
+        builder.Append(8);
+        builder.Append(9);
+        builder.Append(10);
+        std::shared_ptr<arrow::Array> array = *builder.Finish();
+
+        struct ArrowSchema* c_schema = reinterpret_cast<struct ArrowSchema*>(c_schema_ptr);
+        auto c_schema_status = arrow::ExportType(*array->type(), c_schema);
+        if (!c_schema_status.ok()) c_schema_status.Abort();
+
+        struct ArrowArray* c_array = reinterpret_cast<struct ArrowArray*>(c_array_ptr);
+        auto c_array_status = arrow::ExportArray(*array, c_array);
+        if (!c_array_status.ok()) c_array_status.Abort();
+    }
+
+**Java Side**
+
+For this example, we will use `JavaCPP`_ to call our main C++ **fillCArray** function from Java,
+without writing JNI bindings ourselves.
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <project xmlns="http://maven.apache.org/POM/4.0.0"
+             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+             xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+        <modelVersion>4.0.0</modelVersion>
+
+        <groupId>org.example</groupId>
+        <artifactId>java-cdata-example</artifactId>
+        <version>1.0-SNAPSHOT</version>
+
+        <properties>
+            <maven.compiler.source>8</maven.compiler.source>
+            <maven.compiler.target>8</maven.compiler.target>
+            <arrow.version>8.0.0</arrow.version>
+        </properties>
+        <dependencies>
+            <dependency>
+                <groupId>org.bytedeco</groupId>
+                <artifactId>javacpp</artifactId>
+                <version>1.5.7</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.arrow</groupId>
+                <artifactId>arrow-c-data</artifactId>
+                <version>${arrow.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.arrow</groupId>
+                <artifactId>arrow-memory-netty</artifactId>
+                <version>${arrow.version}</version>
+            </dependency>
+        </dependencies>
+    </project>
+
+.. code-block:: java
+
+    import org.bytedeco.javacpp.annotation.Platform;
+    import org.bytedeco.javacpp.annotation.Properties;
+    import org.bytedeco.javacpp.tools.InfoMap;
+    import org.bytedeco.javacpp.tools.InfoMapper;
+
+    @Properties(
+            target = "CDataJavaToCppExample",
+            value = @Platform(
+                    include = {
+                            "CDataCppBridge.cpp"
```

Comment:
Hmm. We should have separate header/C++ files then, we shouldn't include a source file.

Reply:
Sorry, my mistake. Base on this [reference](https://github.com/bytedeco/javacpp#accessing-native-apis) we declare the file *.h as a header and define function FillInt64Array to export.